extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php

```diff
@@ -229,6 +230,7 @@ public function getAllAdminsData() {
 	 */
 	public function getRecentlyJoinedData() {
 		$recentlyJoined = $this->usersModel->getRecentlyJoinedUsers();
+		$recentlyJoined = $this->applyBadges( $recentlyJoined );
```

Comment:
have you considered applying badges on lower level, in the same place where we get info about avatar, here: https://github.com/Wikia/app/blob/2a58df6fa50b7b0e6c53a67a21ba9815ba140a48/extensions/wikia/CommunityPage/models/CommunityPageSpecialUsersModel.class.php#L277 ? Same goes for `getAllMembersData`. This would avoid looping over array 2 times and creating user objects 2 times